sys-libs/libcxx/libcxx-9999.ebuild

```diff
@@ -9,13 +9,13 @@ EGIT_REPO_URI="http://llvm.org/git/libcxx.git
 	https://github.com/llvm-mirror/libcxx.git"
 CMAKE_MIN_VERSION=3.4.3
 
-[ "${PV%9999}" != "${PV}" ] && SCM="git-r3" || SCM=""
+[[ ${PV} == 9999 ]] && SCM="git-r3" || SCM=""
```

Comment:
> Well, I can write these 4 steps in 4 lines of C with O(|input|) complexity. I doubt you can say the same for pattern matching.

But you don't, and you instead rely on poorly reimplementing C in bash. You're not dealing with compiled code, so you also need to count on increased lexing complexity, word splitting, pattern matching etc.

> If we paid more attention to bash code at global scope, maybe metadata generation or emerge -uND world wouldnt be so slow :)

I agree. This conditional should really be removed since you know its result up front, and it only clutters the ebuild and makes the metadata regen slow.

> (I didnt check, but if bash processes endswith as a pattern then there is room for improvement there)

It has to because the argument to `%` **is a pattern**.